### PR TITLE
feature: upgraded to php 8.1

### PR DESCRIPTION
**Pull request by tobybatch**

### CODE REVIEW

**Comment by kevinpapst** [comment]
I know that 8.1 works fine. 

But regarding this PR I wonder if `php:8.1-rc` is a good idea. Why release candidate?
Isn't there an image with a fixed version?

File: Dockerfile [hunk 1]
```diff
@@ -70,7 +71,7 @@ RUN apk add --no-cache \
 
 
 # apache debian php extension base
-FROM php:8.0.14-apache-buster AS apache-php-ext-base
+FROM php:8.1-rc-apache-buster AS apache-php-ext-base
```
Comment (tobybatch) [inline Dockerfile]:
```suggestion
FROM php:8.1-apache-buster AS apache-php-ext-base
```

File: Dockerfile [hunk 2]
```diff
@@ -148,7 +149,7 @@ HEALTHCHECK --interval=20s --timeout=10s --retries=3 \
 # apache base build
 ###########################
 
-FROM php:8.0.14-apache-buster AS apache-base
+FROM php:8.1-rc-apache-buster AS apache-base
```
Comment (tobybatch) [inline Dockerfile]:
```suggestion
FROM php:8.1-apache-buster AS apache-php-ext-base
```

**Comment by tobybatch** [comment]
It was just the latest version, you're probably right.

**Review by kevinpapst — APPROVED** [review]
Not that I understand where the version `20210902` is coming from, but otherwise LGTM 🚀

File: Dockerfile [hunk 3]
```diff
@@ -148,7 +149,7 @@ HEALTHCHECK --interval=20s --timeout=10s --retries=3 \
 # apache base build
 ###########################
 
-FROM php:8.0.14-apache-buster AS apache-base
+FROM php:8.1-apache-buster AS apache-php-ext-base
```
Comment (kevinpapst) [inline Dockerfile:152]: out of curiosity: does renaming the base image change anything? might someone rely on that name? is it linked somewhere in the build process?

File: Dockerfile [hunk 4]
```diff
@@ -199,24 +200,25 @@ COPY --from=composer /usr/bin/composer /usr/bin/composer
...
 # PHP extension intl
 COPY --from=php-ext-intl /usr/local/etc/php/conf.d/docker-php-ext-intl.ini /usr/local/etc/php/conf.d/docker-php-ext-intl.ini
-COPY --from=php-ext-intl /usr/local/lib/php/extensions/no-debug-non-zts-20200930/intl.so /usr/local/lib/php/extensions/no-debug-non-zts-20200930/intl.so
+COPY --from=php-ext-intl /usr/local/lib/php/extensions/no-debug-non-zts-20210902/intl.so /usr/local/lib/php/extensions/no-debug-non-zts-20210902/intl.so
+# PHP extension redis
 COPY --from=php-ext-redis /usr/local/etc/php/conf.d/docker-php-ext-redis.ini /usr/local/etc/php/conf.d/docker-php-ext-redis.ini
-COPY --from=php-ext-redis /usr/local/lib/php/extensions/no-debug-non-zts-20200930/redis.so /usr/local/lib/php/extensions/no-debug-non-zts-20200930/redis.so
+COPY --from=php-ext-redis /usr/local/lib/php/extensions/no-debug-non-zts-20210902/redis.so /usr/local/lib/php/extensions/no-debug-non-zts-20210902/redis.so
 
 ENV DATABASE_URL=sqlite:///%kernel.project_dir%/var/data/kimai.sqlite
```
Comment (kevinpapst) [inline Dockerfile:223]: BTW: that sqlite reference is wrong

Comment (tobybatch) [reply Dockerfile:223]: I don't think it matters, it's only there so we can run `composer install` with out erroring

**Pull request merged**